### PR TITLE
UX: Show a "Summarize" button in topic timeline

### PR DESCRIPTION
**Pull request by pmusaraj**
Makes it a little easy to access the summary feature of large topics,
given that the existing button at the bottom of the OP is easy to miss.

image

This is behind a site setting (default off).

### CODE REVIEW

File: config/locales/client.en.yml [hunk 1]
```diff
@@ -1819,6 +1819,8 @@ en:
       description_time_MF: "There {replyCount, plural, one {is <b>#</b> reply} other {are <b>#</b> replies}} with an estimated read time of <b>{readingTime, plural, one {# minute} other {# minutes}}</b>."
       enable: "Summarize This Topic"
       disable: "Show All Posts"
+      short_label: "Summarize"
+      short_title: "Show a summary of this topic as determined by the community"
```
Comment (eviltrout) [inline config/locales/client.en.yml:1823]:
```suggestion
      short_title: "Show a summary of this topic: the most interesting posts as determined by the community"
```

File: app/assets/javascripts/discourse/app/widgets/topic-timeline.js [hunk 2]
```diff
@@ -351,6 +351,22 @@ createWidget("timeline-footer-controls", {
...
 
+    if (
+      this.siteSettings.summary_timeline_button &&
+      !fullScreen &&
+      topic.has_summary &&
+      !topic.postStream.summary
+    ) {
+      controls.push(
+        this.attach("button", {
+          className: "show-summary btn-small",
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/widgets/topic-timeline.js:362]: We should put an icon here. I feel like our buttons look weird when we don't have one! Does the summarize button in the topic map have one?

**Review by eviltrout — COMMENTED** [review]
This is a neat idea. We should give it a try.

**Review by eviltrout — APPROVED** [review]

File: app/assets/javascripts/discourse/app/widgets/topic-timeline.js (hunk 2 above)
Comment (pmusaraj) [reply app/assets/javascripts/discourse/app/widgets/topic-timeline.js:362]: No, we don't use an icon there. It's a little tricky to find an appropriate one. 

Maybe https://fontawesome.com/v5.15/icons/layer-group?style=solid or https://fontawesome.com/v5.15/icons/stream?style=solid?

Comment (pmusaraj) [reply app/assets/javascripts/discourse/app/widgets/topic-timeline.js:362]: Here's what this look like with the layer-group icon added to both buttons: 

image

Comment (eviltrout) [reply app/assets/javascripts/discourse/app/widgets/topic-timeline.js:362]: That's a really nice icon!

**Comment by CvX** [comment]
I wonder how this will look in more _verbose_ languages.

**Comment by pmusaraj** [comment]
@CvX this is default off. But yeah, it probably won't look great on verbose languages.

**Pull request merged**